### PR TITLE
Export/Import user generative data - alternative

### PR DESCRIPTION
**Pull request by DivyaniPatel96**
Work on story id: 172387
1. export/import backup file support json format
2. encrypt/decrypt json file

### CODE REVIEW

File: app/assets/translations/appOfflineData/apiConstants.ts [hunk 1]
```diff
@@ -14,10 +14,12 @@ export const videoTypeVimeo = "vimeo";
 export const videoTypeYoutube = "youtube";
 export const videoTypeImage = "novideo";
 export const backupGDriveFolderName = 'ParentBuddy';
-export const backupGDriveFileName = 'my.backup';
+export const backupGDriveFileName = 'mybackup.json';
+export const encryptionsKey = 'f06d3a055c7066de31d6d1ae583d7bd18d99840bc74d14aaed63860054004f15';
+export const encryptionsIVKey = 'cd6854fab61ec032563b2e0c38b30c39';
```
Comment (Akhror) [inline app/assets/translations/appOfflineData/apiConstants.ts:19]: Let's not store keys on public repo, get it from configs.

File: app/screens/ChildSetup.tsx [hunk 2]
```diff
@@ -183,16 +187,30 @@ const ChildSetup = ({ navigation }: Props): any => {
   };
   const importFromFile = async (): Promise<any> => {
     if (Platform.OS == "android") {
```
Comment (Akhror) [inline app/screens/ChildSetup.tsx:189]: Can you refactor code, something like:

// Call the appropriate import function based on the platform
if (Platform.OS === 'android') {
  importDataAndroid();
} else {
  importDataNonAndroid();
}

File: app/screens/ChildSetup.tsx [hunk 3]
```diff
@@ -183,16 +187,30 @@ const ChildSetup = ({ navigation }: Props): any => {
...
+              return text;
+            })
+            .catch((error: any) => {
+              console.log("Decrypted error", error);
+              throw error;
+            });
+          const importedJsonData = JSON.parse(await decryptedData);
+          await RNFS.writeFile(tempRealmFile, JSON.stringify(decryptedData), "utf8");
+          oldChildrenData = importedJsonData;
+        } else {
```
Comment (Akhror) [inline app/screens/ChildSetup.tsx:206]: Should we validate file types? How does app behave if user selects inappropriate file?

File: app/screens/ChildSetup.tsx [hunk 4]
```diff
@@ -204,7 +222,9 @@ const ChildSetup = ({ navigation }: Props): any => {
           navigation.navigate('ChildImportSetup', {
             importResponse: JSON.stringify(oldChildrenData)
           });
-          importedrealm.close();
+          if (dataset.name.endsWith('.backup')) {
```
Comment (Akhror) [inline app/screens/ChildSetup.tsx:225]: if (importedrealm) 
check is probably better choice.

File: app/screens/ChildSetup.tsx (hunk 2 above)
Comment (DivyaniPatel96) [reply app/screens/ChildSetup.tsx:189]: okay @Akhror sure, I will refactor above code

File: app/screens/ChildSetup.tsx [hunk 5]
```diff
@@ -181,90 +184,132 @@ const ChildSetup = ({ navigation }: Props): any => {
...
+              Realm.deleteFile({ path: tempRealmFile });
+            } catch (error) {
+              //console.log(error);
             }
 
           }
+          else {
+            setLoading(false);
+            setIsImportRunning(false);
+          }
```
Comment (Akhror) [inline app/screens/ChildSetup.tsx:301]: Let's please introduce a new method to combine lines 218-238 and 280-301

Something like:

```
function handleImportedData(importedData: any) {
  if (importedData.length > 0) {
    userRealmCommon.openRealm()
      .then(() => userRealmCommon.deleteAllAtOnce())
      .then(() => {
        setLoading(false);
        setIsImportRunning(false);
        navigation.navigate('ChildImportSetup', {
          importResponse: JSON.stringify(importedData),
        });
      })
      .catch((error) => console.error('Error handling imported data:', error));

    if (tempRealmFile.endsWith('.backup')) {
      Realm.deleteFile({ path: tempRealmFile });
    }
  } else {
    setLoading(false);
    setIsImportRunning(false);
  }
}
```

**Pull request merged**